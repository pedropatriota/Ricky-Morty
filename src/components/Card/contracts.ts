export interface ICardProps {
	name: string;
	species: string;
	status: string;
	gender: string;
	image: string;
}
