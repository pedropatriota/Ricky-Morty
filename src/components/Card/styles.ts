import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	flex-direction: column;
	width: 18rem;
`;

export const ImagContainer = styled.div<{ image: string }>`
	background-image: url(${props => props.image});
	background-repeat: no-repeat;
	background-size: cover;
	width: 100%;
	max-width: 18rem;
	height: 18rem;
	border-radius: 1rem 1rem 0 0;
`;

export const InfoContainer = styled.div`
	padding: 0.5rem;
	background-color: ${props => props.theme.colors.el};
	border: 1px solid ${props => props.theme.colors.el};
	border-radius: 0 0 1rem 1rem;

	h2 {
		margin-bottom: 1rem;
		color: ${props => props.theme.colors.text};
	}
	p {
		margin-bottom: 0.4rem;
		color: ${props => props.theme.colors.text};
	}
`;
