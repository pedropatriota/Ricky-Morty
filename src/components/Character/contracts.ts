export interface ICharacterProps {
	id: number;
	name: string;
	species: string;
	status: string;
	gender: string;
	image: string;
}
