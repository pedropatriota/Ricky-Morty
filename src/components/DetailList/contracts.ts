export type UnknownObject = Record<string, string | number>;

export interface IDetailListProps<T extends UnknownObject> {
	dataArr: Array<Partial<T>>;
	params: { label: string; value: string }[];
}
