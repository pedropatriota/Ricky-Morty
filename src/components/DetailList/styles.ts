import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	gap: 0.75rem;
	flex-direction: column;
	margin-bottom: 0.75rem;
`;
