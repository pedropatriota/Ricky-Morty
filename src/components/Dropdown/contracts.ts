import { ActionMeta, GroupBase, SingleValue } from 'react-select';

type Filter = SingleValue<{ label: string; value: string }>;

export interface IDropdownProps {
	options: readonly (Filter | GroupBase<{ label: string; value: string }>)[];
	filter: Filter;
	label?: string;
	handleFilter: (
		newValue: Filter | unknown,
		actionMeta: ActionMeta<Filter | unknown>
	) => void;
}
