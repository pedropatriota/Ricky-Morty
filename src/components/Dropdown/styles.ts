import styled from 'styled-components';
import Select from 'react-select';

export const ReactSelect = styled(Select)`
  .react-select__control {
    width: min(90vw, 12.5rem);
    background-color: ${(props) => props.theme.colors.el};
    border-color: ${(props) => props.theme.colors.border};
    box-shadow: 0 0 0 transparent;
    height: 3.375rem;

    &:hover {
      border-color: ${(props) => props.theme.colors.border};
    }
  }

  .react-select__single-value {
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__menu {
    width: 12.5rem;
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__option,
  .react-select__option--is-focused 
  {
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__dropdown-indicator,
  .react-select__indicator-separator {
    display: none;
  }
`;
