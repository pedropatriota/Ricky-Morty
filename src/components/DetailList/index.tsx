import { renderData } from '../../utils';
import { Fragment, useId } from 'react';
import { Container } from './styles';
import type { IDetailListProps, UnknownObject } from './contracts';

const DetailList = ({ dataArr, params }: IDetailListProps<UnknownObject>) => {
	const id = useId();
	return (
		<div>
			{dataArr.map(() => {
				return (
					<Container key={id}>
						{params?.map(({ label, value }) => {
							return (
								<Fragment key={value}>
									{renderData(label, value)}
								</Fragment>
							);
						})}
					</Container>
				);
			})}
		</div>
	);
};

export default DetailList;
