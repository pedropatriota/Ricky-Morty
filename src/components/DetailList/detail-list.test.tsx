import { screen } from '@testing-library/react';
import DetailList from '.';
import { renderWithProviders } from '../../utils';
import { UnknownObject } from './contracts';
import { light } from '../../styles/themes';

describe('DetailListComponent', () => {
	const dataArr: Partial<UnknownObject>[] = [
		{
			id: 1,
			name: 'Rick Sanchez',
			image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
			species: 'Human',
			status: 'Alive',
			gender: 'Male'
		}
	];
	const params = [
		{
			label: 'Species',
			value: 'Human'
		},
		{
			label: 'Status',
			value: 'Alive'
		},
		{
			label: 'Gender',
			value: 'Male'
		}
	];
	it('should render correctly', () => {
		renderWithProviders(<DetailList dataArr={dataArr} params={params} />, light);

		const characterProps = screen.getAllByText(/human|alive|male/i);

		expect(characterProps).toHaveLength(3);
	});

	it('should match the snapshot', () => {
		const { container } = renderWithProviders(
			<DetailList dataArr={dataArr} params={params} />,
			light
		);

		expect(container).toMatchSnapshot();
	});
});
