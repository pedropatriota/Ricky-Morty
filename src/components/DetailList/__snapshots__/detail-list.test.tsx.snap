// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailListComponent should match the snapshot 1`] = `
<div>
  <div>
    <div
      class="sc-aXZVg enkuQI"
    >
      <p>
        <strong>
          Species
          : 
        </strong>
         
        Human
      </p>
      <p>
        <strong>
          Status
          : 
        </strong>
         
        Alive
      </p>
      <p>
        <strong>
          Gender
          : 
        </strong>
         
        Male
      </p>
    </div>
  </div>
</div>
`;
